$OpenBSD: patch-configure,v 1.5 2017/09/16 22:42:37 jeremy Exp $

Fix so name, checking for DOT and DOXYGEN.

Override the arch setting to remove OpenBSD version from it,
so ports don't have to be bumped when OpenBSD version changes.

Rename HEAP_ALIGN_LOG to HEAP_PAGE_ALIGN_LOG to make ruby use
operating system page size as GC heap page size (as was true
in earlier ruby versions).

Index: configure
--- configure.orig
+++ configure
@@ -424,7 +424,7 @@ if test -z "$2"; then :
 
 else
 
-	$as_echo_n "   * $1:                     " | head -c26
+	$as_echo_n "   * $1:                     " | dd bs=26 count=1 status=none
 	if test "$FOLD"; then :
 
 	    echo "$2" | fold -s -w50 |
@@ -19573,12 +19573,12 @@ fi
     colorize_result "$rb_cv_page_size_log" ; }
   if test $rb_cv_page_size_log != no; then
     cat >>confdefs.h <<_ACEOF
-#define HEAP_ALIGN_LOG $rb_cv_page_size_log
+#define HEAP_PAGE_ALIGN_LOG $rb_cv_page_size_log
 _ACEOF
 
   else
     cat >>confdefs.h <<_ACEOF
-#define HEAP_ALIGN_LOG 12
+#define HEAP_PAGE_ALIGN_LOG 12
 _ACEOF
 
   fi
@@ -25645,7 +25645,7 @@ esac
   openbsd*|mirbsd*) :
 
 	SOLIBS='$(LIBS)'
-	LIBRUBY_SO='lib$(RUBY_SO_NAME).so.$(MAJOR).'`expr ${MINOR} \* 10 + ${TEENY}`
+	LIBRUBY_SO='lib$(RUBY_SO_NAME).so.'${LIBruby24_VERSION}
 	 ;; #(
   solaris*) :
 
@@ -27004,7 +27004,7 @@ _ACEOF
 _ACEOF
 
 else
-    arch="${target_cpu}-${target_os}"
+    arch="${target_cpu}-openbsd"
     case "$arch" in #(
   le32-nacl) :
     arch="pnacl" ;; #(
